sql/chunk_constraint.sql

```diff
@@ -0,0 +1,174 @@
+-- Creates a constraint on a chunk.
+CREATE OR REPLACE FUNCTION _timescaledb_internal.chunk_constraint_add_table_constraint(
+    chunk_constraint_row  _timescaledb_catalog.chunk_constraint
+)
+    RETURNS VOID LANGUAGE PLPGSQL AS
+$BODY$
+DECLARE
+    sql_code    TEXT;
+    chunk_row _timescaledb_catalog.chunk;
+    hypertable_row _timescaledb_catalog.hypertable;
+    constraint_oid OID;
+    def TEXT;
+BEGIN
+    SELECT * INTO STRICT chunk_row FROM _timescaledb_catalog.chunk c WHERE c.id = chunk_constraint_row.chunk_id;
```

Comment:
Fair enough. FWIW, I don't think this is a PLPGSQL thing but rather an interface-design thing. I prefer to have functions that are decoupled interface wise and try to avoid passing to much context from one function to another unless the code is performance sensitive (Thus I cringe at a lot of the context stuff in C but understand it's necessary). We should probably add some more helper functions however and I am all up for ideas on creating better PLPGSQL code (one idea I had was to make better use of views). But let's leave for another PR.

Reply:
For the record, I never stated this as a language issue or C vs PLPGSQL issue. It is simply a pattern in _our_ code. Passing around row types as suggested would go a long way to resolve the issue. I see no benefit of throwing away context information for the sake of keeping interfaces "clean", and I further fail to see how it makes things cleaner to always pass a `chunk_id` instead of e.g., a `chunk_row` as it requires further code bloat in the target function. FWIW, the context stuff in C is a completely unrelated issue as it mostly a way to pass different type of data to (callback) functions that has the same generic interface. But this is completely besides the point I am making here. Let's keep this discussion in mind for the future.